404.html

```diff
@@ -0,0 +1,20 @@
+<!doctype html>
+<html>
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <meta name="apple-mobile-web-app-capable" content="yes">
+    <meta name="mobile-web-app-capable" content="yes">
+    <link rel="icon" type="image/x-icon" href="favicon.ico">
+    <script>
+        const serverSlash = 5;
+        const route = location.href.split("/");
+        route.splice(0,serverSlash);
+        const path = `${location.href.split("/").slice(0,serverSlash).join("/")}${location.search? location.search + '&' : '?'}nextRoute=${route}`;
```

Comment:
Please refactor the "magic" to be more readable (e. g. additional variables with understandable name, maybe new methods, etc.